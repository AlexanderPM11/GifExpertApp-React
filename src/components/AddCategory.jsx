import { useState } from "react"
export const AddCategory = ({ onAddCategory }) => {


    const [inputValue, setinputValue] = useState('One Punch');

    // Me quede en el punto de emitir evento al padre

    const onInputChange = (event) => {
        setinputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length === 0) return
        onAddCategory(inputValue.trim())
        setinputValue('')
    }
    return (

        <form onSubmit={onSubmit}>
            <input
                type="text"
                value={inputValue}
                placeholder="Buscar Gifs"
                onChange={onInputChange}

            />
        </form>
    )
}