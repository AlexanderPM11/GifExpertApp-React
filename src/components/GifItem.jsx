
export const GitItem = (gif) => {
    return (
        <div className="card">
            <img className="img-custom" src={gif.gif.url} alt={gif.gif.title} />

            <p>{gif.gif.title}</p>
        </div>
    );
};
