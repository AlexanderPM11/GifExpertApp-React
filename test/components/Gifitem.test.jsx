import { render, screen } from '@testing-library/react';
import { GitItem } from "../../src/components/Components";

describe('Test in <GitItem />', () => {
    const gifData = {
        url: 'url_de_ejemplo',
        title: 'TÃ­tulo de ejemplo'
    };

    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<GitItem gif={gifData} />);
        expect(container).toMatchSnapshot();
    });

    // eslint-disable-next-line no-undef
    test('Probar la alt de la img', () => {

        render(<GitItem gif={gifData} />);
        // screen.debug();

        const { src, alt } = screen.getByRole('img');
        expect(alt).toBe(gifData.title);
    });


    // eslint-disable-next-line no-undef
    test('Mostrar titulo en el componente', () => {
        render(<GitItem gif={gifData} />);
        expect(screen.getByText(gifData.title)).toBeTruthy();
    });
});
