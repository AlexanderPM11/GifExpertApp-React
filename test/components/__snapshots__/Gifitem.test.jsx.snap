// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GitItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Título de ejemplo"
      class="img-custom"
      src="url_de_ejemplo"
    />
    <p>
      Título de ejemplo
    </p>
  </div>
</div>
`;
